### PR TITLE
adding support for broker metadata as per new osb version

### PR DESCRIPTION
**Pull request by jayeshjk**
This PR adds support for adding broker metadata for service instances. This is as per OSB Spec [here](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#service-instance-metadata).
As of now, Service Fabrik doesn't need to add any specific metadata. Further this metadata (if we decide to add it in future) is returned in the fetch service instance call.

### CODE REVIEW

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-1 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-2 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

File: broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js [hunk 1]
```diff
@@ -631,6 +631,11 @@ class ServiceBrokerApiController extends FabrikBaseController {
             }
             _.set(body,'service_id',_.get(resource, 'spec.serviceId'));
             _.set(body,'plan_id',_.get(resource, 'spec.planId'));
+            
+            if(_.has(resource, 'spec.metadata')) {
+              _.set(body, 'matadata', _.get(resource, 'spec.metadata'));
```
Comment (anoopjb) [inline broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js:636]:
```suggestion
              _.set(body, 'metadata', _.get(resource, 'spec.metadata'));
```
Minor typo.

Comment (jayeshjk) [reply broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js]: Thanks @anoopjb for noting this. Will add a UT for this for better visibility.

**Review by vivekzhere — APPROVED** [review]

File: broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js (hunk 1 above)
Comment (jayeshjk) [reply broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js]: Added UT.

**Pull request merged**